spec/graphql/analysis/ast_spec.rb

```diff
@@ -71,6 +71,17 @@ def result
     end
   end
 
+  class AstArguments < GraphQL::Analysis::AST::Analyzer
+    def on_enter_argument(node, parent, visitor)
+      @argument = visitor.argument_definition
```

Comment:
Is `@argument` here the same as `node`? 

If so, that's basically why it was previously implemented this way. `visitor.argument_definition` was not the _current_ arg defn, but rather, the argument definition which _owns_ the value we're looking at right now. 

If `@argument` _is_ `node`, then it's not clear to me why we need the new method. 

But maybe it's something else?

Reply:
Do you mean same argument in the schema context? If so, yes it's the same argument. Of course `node` is a `GraphQL::Language::Nodes::Argument` and `@argument` is a `GraphQL::Argument`.

It's equivalent to a field with `node` and `visitor.field_definition` now.

I'm trying to refresh my memory on this... but basically without this addition `visitor.argument_definition` wouldn't be the same argument as `node` but its "owner" (if it exists) and you couldn't even easily get the current argument definition.

Reply:
ohhh right, `node` is a Node and `argument_definition` is a definition. Also refreshing my memory apparently 😆 . Thanks for clarifying.